v2/dtos/intervalaction.go

```diff
@@ -8,48 +8,40 @@ package dtos
 import (
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
 	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/models"
+
+	"github.com/google/uuid"
 )
 
 // IntervalAction and its properties are defined in the APIv2 specification:
 // https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-scheduler/2.x#/IntervalAction
 type IntervalAction struct {
 	common.Versionable `json:",inline"`
-	Created            int64  `json:"created,omitempty"`
-	Modified           int64  `json:"modified,omitempty"`
-	Id                 string `json:"id,omitempty" validate:"omitempty,uuid"`
-	Name               string `json:"name" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
-	IntervalName       string `json:"intervalName" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
-	Protocol           string `json:"protocol,omitempty"`
-	Host               string `json:"host,omitempty"`
-	Port               int    `json:"port,omitempty"`
-	Path               string `json:"path,omitempty"`
-	Parameters         string `json:"parameters,omitempty"`
-	HTTPMethod         string `json:"httpMethod,omitempty" validate:"omitempty,oneof='GET' 'HEAD' 'POST' 'PUT' 'DELETE' 'TRACE' 'CONNECT'"`
-	User               string `json:"user,omitempty"`
-	Password           string `json:"password,omitempty"`
-	Publisher          string `json:"publisher,omitempty"`
-	Target             string `json:"target" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
-	Topic              string `json:"topic,omitempty"`
+	Created            int64   `json:"created,omitempty"`
+	Modified           int64   `json:"modified,omitempty"`
+	Id                 string  `json:"id,omitempty" validate:"omitempty,uuid"`
+	Name               string  `json:"name" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
+	IntervalName       string  `json:"intervalName" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
+	Address            Address `json:"address" validate:"required"`
```

Comment:
The 'dive' only works for slice and map, see https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Dive

In this case, the request DTO looks like:
```
  {
    "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
    "apiVersion":"v2",
    "action": {
        "apiVersion":"v2",
        "name": "query",
        "intervalName": "afternoon",
        "address":{
            "type": "REST",
            "host": "123.123.0.1",
            "port": 123,
            "httpMethod": "GET"
        }
            
    }
  }
```
Actually I can success to verify the `Address` without call validate on Address. But I need to verify the REST or MQTT protocol properties, see https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/address.go#L25.

Otherwise we can
1. Migrate the v1 validator to verify the deeper struct https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/models/validator.go
2. Or write a custom validation tag at https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/validator.go

Reply:
Doesn't this validation attribute handle it?
```
validate:"oneof='REST' 'MQTT'"`
```